test/integration/api/authentication.spec.js

```diff
@@ -899,6 +1076,140 @@ test.serial('performs an OpenID authentication', async (t) => {
   await browser.close()
 })
 
+if (shouldExecuteBuiltInSSOTest) {
+  if (getOAuthConfiguration('github')) {
+    // Must run serially because config is updated
+    test.serial('Github OAuth authentication works', async (t) => {
+      const provider = 'github'
+
+      const config = getOAuthConfiguration(provider)
+      if (!config) return t.pass()
+
+      const { credentials, ssoConnection } = config
+
+      const { username, password } = credentials
+
+      const executeBrowserScenario = async ({ page }) => {
+        // trigger the SSO authentication (user clicks on the SSO button)
+        await page.goto(`${apiServerUrl}/auth/sso/${publicPlatformId}/${provider}`)
+
+        const usernameSelector = 'input[name="login"]'
+
+        await page.waitForSelector(usernameSelector)
+
+        // fill login form
+        await page.type(usernameSelector, username)
+        await page.type('input[type="password"]', password)
+        await page.click('input[type="submit"]')
+
+        // authorize form is displayed the first time
+        const authorizeButtonSelector = 'button[name="authorize"]'
+        const authorizeButtonExists = await checkElementVisible(page, authorizeButtonSelector)
+        if (authorizeButtonExists) {
+          // wait a little bit before clicking the button as it can be disabled at first
+          await new Promise(resolve => setTimeout(resolve, 1000))
```

Comment:
I suppose there is no mean to wait for a given state of the button using puppeteer? (like what you can do to wait for an element to show up)

Reply:
I'll try to make it more robust with `.waitForFunction()` which can evaluate an expression (element visibility and not disabled if it's a button).
https://github.com/puppeteer/puppeteer/blob/v5.0.0/docs/api.md#framewaitforfunctionpagefunction-options-args